src/tracks/playabletrack/wavetrack/ui/WaveClipTrimHandle.cpp

```diff
@@ -89,5 +131,15 @@ UIHandle::Result WaveClipTrimHandle::Release
 
 UIHandle::Result WaveClipTrimHandle::Cancel(AudacityProject* pProject)
 {
-    return RefreshCode::RefreshNone;
+    switch (mTargetBorder)
+    {
+    case Border::Left:
+        mClip->TrimLeftTo(mInitialEdgeTime);
+        break;
+    case Border::Right:
+        mClip->TrimRightTo(mInitialEdgeTime);
+        break;
+    default: break;
+    };
+    return RefreshCode::RefreshCell;
```

Comment:
The Cancel procedure is good.

The Release is still empty.  It will require pushing an undo history item.